src/spine.js

```diff
@@ -127,15 +85,37 @@ pc.extend(pc, function () {
 
         this._position = new pc.Vec3();
 
-        var atlas = new spine.TextureAtlas(atlasData, function (path) {
-            return new SpineTextureWrapper(textureData[path]);
-        });
+        var atlas;
+
+        // EP: As instructed: https://github.com/playcanvas/playcanvas-spine/pull/73
+        //
+        // API differs before we can know which which file version this is supposed to support.
+        // The data file opened will determine which code paths are used.  This file may
+        // or may not be concatenated with the spine library that supports it.
+        //
+        // use the length of the function signatures to guess which library has been concatenated.
+        if (spine.TextureAtlas.length === 1) {
+            // spine 4.1
+            atlas = new spine.TextureAtlas(atlasData);
+            for (const page of atlas.pages) {
+                page.setTexture(new SpineTextureWrapper(textureData[page.name]));
+            }
+        } else {
+            // spine 3.6 and 3.8
+            atlas = new spine.TextureAtlas(atlasData, function (path) {
+                return new SpineTextureWrapper(textureData[path]);
+            });
+        }
+
         var json = new spine.SkeletonJson(new spine.AtlasAttachmentLoader(atlas));
         json.scale *= 0.01;
         var _skeletonData = json.readSkeletonData(skeletonData);
-        this.skeletonVersion = _skeletonData.version;
-        this._spine_3_6_0 = versionCompare(this.skeletonVersion, "3.6.0") <= 0;
-        this._spine_3_7_99 = versionCompare(this.skeletonVersion, "3.7.99") <= 0;
+
+        // Compatibility queries
+        this.skeletonVersion = semver.valid(semver.coerce(_skeletonData.version));
+        this._spine_3_6_0 = semver.satisfies(this.skeletonVersion, '<3.6.0'); // version 3.6.0 or below
+        this._spine_3_7_99 = semver.satisfies(this.skeletonVersion, '<3.8.0'); // version 3.7.99 or below
```

Comment:
And should this be:
`<=3.7.99`

Reply:
need to handle the case of `3.7.100` etc.  I'll update the comment.

Reply:
Here's the thing on this one.  I can't find where Spline editor ever produced files with versions this high, so I took this to mean "upper limit before 3.8".  Let's go with your suggestion.

Reply:
completed in 29b6250